sql/table.h

```diff
@@ -769,28 +769,28 @@ struct TABLE_SHARE
 
   /**
     System versioning support.
-   */
-   struct period_info_t
-   {
-     TABLE_SHARE *s;
-     uint16 start_fieldno;
-     uint16 end_fieldno;
-     Lex_ident name;
-
-     Field *start_field()
-     {
-       return s->field[start_fieldno];
-     }
-     Field *end_field()
-     {
-       return s->field[end_fieldno];
-     }
-   };
+  */
+  struct period_info_t
+  {
+    uint16 start_fieldno;
+    uint16 end_fieldno;
+    Lex_ident name;
+    Field *start_field(TABLE_SHARE *s) const
+    {
+      return s->field[start_fieldno];
```

Comment:
Then it's better to pass Field**, so it will work with both TABLE_SHARE and TABLE

Reply:
`TABLE` always have `TABLE_SHARE` inited.

Reply:
but they have different `field` members instances. You obviously cannot access actual table data through share's fields

Reply:
It was always used via `TABLE_SHARE` (including your code).

Reply:
True. Really, there is no best solution or I don't see one. Each one lacks something, so this is your freedom to choose. But keep away from making data members without good reason.

Reply:
In fact, making accessor just over subscript operator looks silly to me (it looks silly just now anyway).

Reply:
I guess, we are just fine with current variant.